gdb/__init__.py

```diff
@@ -640,6 +640,12 @@ def __str__(self) -> str:
         if type_flags & lldb.eTypeIsPointer:
             return "0x%x" % self._as_number()
 
+        # Check for synthetic children.
+        if (t.GetTypeClass() == lldb.eTypeClassStruct):
+            valstr = str(self._sbvalue_object.GetSyntheticValue())
+            if (valstr != "No value" and
+                valstr.find("$1") == -1):
```

Comment:
Would `SBValue.IsValid()`  work here instead of checking the string against "No value"?

What is the check against "$1" supposed to do? If possible, we should replace this with some API call that tests some property of the SBValue rather than pattern-matching the stringified version. If this is not possible, is it always "$1" or is this the kind of convenience variable that depends on how many expressions you've evaluated before? Can it ever be "$2" or "$3"?

Reply:
The problem comes from 'address.test'.  There is a part of the text that specifically does not want to use a synthetic value (but there is a valid one).  The Check line is
CHECK: Pretty s = pretty MyStruct

What my code returns, without the '$1' test is: 
Pretty s = (MyStruct) $1 = pretty MyStruct

I don't really like the '$1' hack, but I'm not sure how to handle/fix this particular case?

Reply:
Using IsValid does work, allowing me to not text for "No Value"

Reply:
I've been trying some stuff and it looks like lldb is pretty inflexible about what `str(sbvalue)` does. It will always generate a string with the following format
```
(${val.GetType().GetName()}) ${val.GetName()} = ${val.GetSummary()} {
  children
  ...
}
```

In this case, the `"$1"` comes from `SBValue.GetName()`. If we want to match what gdb does, I'm afraid we'll need to implement the logic to generate the string ourselves without relying on `str(val)`. 

Or we could live with `(MyStruct) $1 = pretty MyStruct` and relax address.test to match that output.